plugins/modules/discovery_intent.py

```diff
@@ -548,6 +593,32 @@ def create_params(self, credential_ids=None, ip_address_list=None):
         if credential_ids is None:
             credential_ids = []
 
+        http_read_credential = self.validated_config[0].get('http_read_credential')
+        http_write_credential = self.validated_config[0].get('http_write_credential')
+        if http_read_credential:
+            if not (http_read_credential.get('password') and isinstance(http_read_credential.get('password'), str)):
+                msg = "Http Read Credential must have a password of type string"
```

Comment:
for better clarity
msg = "The password for the HTTP read credential must be of string type."

Reply:
Addressed